liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/applications/DeployMojoSupport.java

```diff
@@ -108,6 +108,29 @@ protected void installLooseConfigWar(MavenProject proj, LooseConfigData config)
                     MessageFormat.format(messages.getString("error.project.not.compile"), proj.getId()));
         }
 
+        if (proj.getProperties().containsKey("container") ||
+            (System.getProperty("container") != null)) {
+            // Set up the config to replace the absolute path names with ${variable}/target type references
+            config.setProjectRoot(proj.getBasedir().getAbsolutePath());
+            config.setSourceOnDiskName("${"+PROJECT_ROOT_NAME+"}");
+            // Set the property to include the variable definition in liberty-plugin-variable-config.xml
+            proj.getProperties().setProperty("liberty.var."+PROJECT_ROOT_NAME, proj.getBasedir().getAbsolutePath());
```

Comment:
This seems late in the process to me to be setting this. I thought the code that handles processing these properties and creates the liberty-plugin-variable-config.xml would have already run. Also, use getCanonicalPath().

Reply:
The code that handles processing these properties and creates the liberty-plugin-variable-config.xml is `StartDebugMojoSupport.copyConfigFiles()`, which has already run in the `deploy` goal.  However for the `dev` goal, copyConfigFiles() is run in DevUtil.getServerTask(), which is called by DevMojo.startServer().  I had incorrectly stated before that serverTask is null; it is not null even for the container case (unless it has not been initialized yet), although its only purpose for containers would be to get the messages log location.  We should use a separate PR to clean up this logic in `dev` to call specifically call copyConfigFiles() and avoid using a serverTask for containers.